instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/AlternativeRouteSelectionTest.kt

```diff
@@ -0,0 +1,205 @@
+import com.mapbox.navigation.instrumentation_tests.utils.routes.MockRoutesProvider
+import com.mapbox.navigation.instrumentation_tests.utils.runOnMainSync
+import com.mapbox.navigation.testing.ui.BaseTest
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import com.mapbox.navigation.ui.maps.camera.NavigationCamera
+import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource
+import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSourceOptions
+import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
+import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLine
+import kotlinx.android.synthetic.main.activity_basic_navigation_view.*
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class AlternativeRouteSelectionTest : BaseTest<BasicNavigationViewActivity>(
+    BasicNavigationViewActivity::class.java
+) {
+
+    @get:Rule
+    val mapboxNavigationRule = MapboxNavigationRule()
+
+    @get:Rule
+    val mockLocationReplayerRule = MockLocationReplayerRule(mockLocationUpdatesRule)
+
+    private lateinit var initIdlingResource: MapStyleInitIdlingResource
+
+    protected lateinit var mockRoute: MockRoute
+
+    protected lateinit var mapboxNavigation: MapboxNavigation
+
+    protected lateinit var routeLineApi: MapboxRouteLineApi
+    protected lateinit var routeLineView: MapboxRouteLineView
+
+    protected lateinit var mapboxNavigationViewportDataSource: MapboxNavigationViewportDataSource
+    protected lateinit var navigationCamera: NavigationCamera
+
+    protected lateinit var navigationLocationProvider: NavigationLocationProvider
+    protected lateinit var locationPlugin: LocationComponentPlugin
+
+    @Before
+    fun setUp() {
+        initIdlingResource = MapStyleInitIdlingResource(activity.mapView)
+        initIdlingResource.register()
+        Espresso.onIdle()
+
+        mockRoute = getRoute(activity)
+        mockWebServerRule.requestHandlers.addAll(mockRoute.mockRequestHandlers)
+
+        val route = mockRoute.routeResponse.routes()[0]
+
+        runOnMainSync {
+            mockLocationUpdatesRule.pushLocationUpdate {
+                latitude = mockRoute.routeWaypoints.first().latitude()
+                longitude = mockRoute.routeWaypoints.first().longitude()
+            }
+            mockLocationReplayerRule.playRoute(route)
+
+            mapboxNavigation = MapboxNavigation(
+                NavigationOptions.Builder(activity)
+                    .accessToken(getMapboxAccessTokenFromResources(activity))
+                    .build()
+            )
+            mapboxNavigation.setRoutes(mockRoute.routeResponse.routes())
```

Comment:
You don't actually need the mock webserver, mock route, and all that setup if your using `setRoutes` (I understand this is a mistake actually copied from the `SimpleMapViewNavigationTest.kt`) because there's no request being made.

Feel free to remove the mock web server setup from this test or to actually call `mapboxNavigation.requestRoutes`.

Reply:
done.